### PR TITLE
dnsdist: Split in two packages `dnsdist` and `dnsdist-full`

### PR DESCRIPTION
**Pull request by rgacogne**
Maintainer: @Habbie 
Compile tested: TP-Link Archer C7 AC1750 v5, 19.07, 21.02 and master
Run tested: TP-Link Archer C7 AC1750 v5, 19.07 and 21.02

Description:

- `dnsdist-full` has all optional features enabled, but is a big package in term of both flash and memory footprint.
- `dnsdist` only keeps the features that make the most sense on embeded devices, but can also be customised to match the user's needs, up to the point where it matches `dnsdist-full`.

Signed-off-by: Remi Gacogne remi.gacogne@powerdns.com

### CODE REVIEW

File: net/dnsdist/Makefile [hunk 1]
```diff
@@ -25,6 +24,35 @@ PKG_CONFIG_DEPENDS:= \
...
+  users while shunting or blocking abusive traffic.
+endef
+
+define Package/dnsdist/conffiles/Default
+/etc/dnsdist.conf
+/etc/config/dnsdist
+/etc/init.d/dnsdist
+endef
+
 define Package/dnsdist/config
```
Comment (BKPepe) [inline net/dnsdist/Makefile:56]: I think, it would be good to move this to the ``Config.in`` as it is getting very long.

Comment (rgacogne) [reply net/dnsdist/Makefile:56]: That makes complete sense to me, will do! Thanks!

Comment (rgacogne) [reply net/dnsdist/Makefile:56]: Done!

**Comment by Habbie** [comment]
I have also tested this (docker, x86_64) and everything looks good to me.

**Pull request merged**